### PR TITLE
refactor: [M3-8462] - Upgrade to TanStack Query v5

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Finishes updating React Query (aka TanStack Query)  to `latest` 📦 
- Fixes deprecations according to [this guide](https://tanstack.com/query/v5/docs/framework/react/guides/migrating-to-v5)

## Main Changes 🔁 
- Mutations use `isPending` instead of `isLoading` to track if a mutation is executing
- The option `keepPreviousData` (an option we use to prevent flickering on paginated endpoints) is no longer a useQuery option. We now use must use `placeholderData` with the `keepPreviousData` helper.
- `initialPageParam` is now a required option on Infinite Queries

## Preview 📷

Screenshot 2024-08-19 at 6 32 32 PM

## How to test 🧪

- Check the diff for anything suspicious 🔍 
- Perform general create/update/delete test for various features of Cloud Manager 🧪 
- Verify all automated testing passes ✅ 

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/queries/account/account.ts [hunk 1]
```diff
@@ -89,13 +90,13 @@ export const useChildAccountsInfiniteQuery = (options: RequestOptions) => {
       }
       return page + 1;
     },
-    keepPreviousData: true,
+    initialPageParam: 1,
```
Comment (jaalah-akamai) [inline packages/manager/src/queries/account/account.ts:93]: Should we use `placeholderData` here? I'll test this part of the Parent/Child code shortly.

File: packages/manager/src/queries/account/queries.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export const accountQueries = createQueryKeys('account', {
         filter: options.filter,
         headers: options.headers,
         params: {
-          page: pageParam,
+          page: pageParam as number,
```
Comment (jaalah-akamai) [inline packages/manager/src/queries/account/queries.ts:62]: Nit: Couldn't we type `pageParam` as `queryFn: ({ pageParam }: { pageParam: number }) =>`? I'm kinda interested why it's defined as `unknown`

File: packages/manager/src/queries/cloudpulse/metrics.ts [hunk 3]
```diff
@@ -36,30 +38,37 @@ export const useCloudPulseMetricsQuery = (
...
+          },
+          {
+            cancelRefetch: true,
+          }
+        );
+      }
+    }
+  }, [query.error]);
+
+  return query;
```
Comment (jaalah-akamai) [inline packages/manager/src/queries/cloudpulse/metrics.ts:71]: cc: @venkymano for transparency - just letting you know of the change

**Review by jaalah-akamai — COMMENTED** [review]
Thank you for the effort in doing this 🚀

File: packages/manager/src/queries/account/account.ts (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/queries/account/account.ts:93]: I don't think we need `keepPreviousData` / `placeholderData` on an infinite query, but I will double check this.

File: packages/manager/src/features/Betas/BetaSignup.tsx [hunk 4]
```diff
@@ -138,7 +138,7 @@ EAP and the MSA, this EAP shall be deemed controlling only with respect to its e
     <>
       <LandingHeader title="Sign Up" />
       <Paper>
-        {isLoading ? (
+        {isLoading || !beta ? (
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/Betas/BetaSignup.tsx:141]: Mutations will use `isPending`, but queries like `useBetaQuery` in this case, can continue to use `isLoading`.

[the remark this answers is not in the document]
Comment (jaalah-akamai) [reply packages/manager/src/features/Betas/BetaSignup.tsx:141]: Great explanation ty

**Review by jaalah-akamai — APPROVED** [review]

File: packages/manager/src/features/Lish/Glish.tsx [hunk 5]
```diff
@@ -10,16 +10,14 @@ import type { LinodeLishData } from '@linode/api-v4/lib/linodes';
 import type { Linode } from '@linode/api-v4/lib/linodes';
 import type { VncScreenHandle } from 'react-vnc';
 
-interface Props {
+interface Props extends LinodeLishData {
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Lish/Glish.tsx:13]:
```suggestion
interface Props extends Omit<LinodeLishData, 'weblish_url'> {
```

File: packages/manager/src/features/Lish/Weblish.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ import { ErrorState } from 'src/components/ErrorState/ErrorState';
 import type { Linode } from '@linode/api-v4/lib/linodes';
 import type { LinodeLishData } from '@linode/api-v4/lib/linodes';
 
-interface Props {
+interface Props extends LinodeLishData {
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Lish/Weblish.tsx:12]:
```suggestion
interface Props extends Pick<LinodeLishData, 'weblish_url' | 'ws_protocols'> {
```

**Review by dwiley-akamai — COMMENTED** [review]
Still a bit more testing and a few more files I want to take a closer look, but it's looking good so far:

CRUD operations for Linodes, Volumes, VPC, Firewalls, Domains, Databases, Kubernetes ✅

**Review by dwiley-akamai — APPROVED** [review]

File: packages/manager/src/features/Users/UserPermissions.tsx [hunk 7]
```diff
@@ -724,7 +724,7 @@ class UserPermissions extends React.Component<CombinedProps, State> {
           });
 
           // Update the user's grants directly in the cache
-          this.props.queryClient.setQueriesData<Grants>(
+          this.props.queryClient.setQueryData<Grants>(
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/Users/UserPermissions.tsx:727]: Yeah exactly. I think `setQueriesData` was accidentally used here

File: packages/manager/src/queries/account/queries.ts (hunk 2 above)
Comment (bnussman-akamai) [reply packages/manager/src/queries/account/queries.ts:62]: Yeah, that would also work. I can't really find a way to make `pageParam ` any more type-safe. It seems like with either solution, it isn't strongly typed in the factory or in the query 🫤

**Pull request merged**